block_model.rb

```diff
@@ -0,0 +1,73 @@
+class BlockModel
+  attr_reader :blocks
+
+  def initialize(blocks)
+    @blocks = blocks
+  end
+
+  def reblock(rx, ry, rz)
+    new_blocks = []
+    nx = @blocks.length / rx
+    ny = @blocks.length[0] / ry
+    nz = @blocks.length[0][0] / rz
+    for i in (0..nx)
+      new_blocks[i] = []
+      for j in (0..ny)
+        new_blocks[i][j] = []
+        for k in (0..nz)
+          new_blocks[i][j][k] = {}
+        end
+      end
+    end
+
+    n_i = 0
+    n_j = 0
+    n_k = 0
+    for i in (0..@blocks.length - 1).step(rx)
+      for j in (0..@blocks[0].length - 1).step(ry)
+        for k in (0..@blocks[0][0].length - 1).step(rz)
+          n = {}
+          for ri in (i..rx-1)
+            for rj in (i..ry-1)
+              for rk in (i..rx-1)
```

Comment:
Is not necessary to have a lot of fors in this function are many forms to solution  this reblock with a few fors(min 3)